File: backend/src/domain/repository/matriculate-repository.ts
import { Matriculate } from "../model/matriculate.ts";

export interface MatriculateRepository {
  enroll(user_id: string, course_id: string): Promise<void>;
  unenroll(user_id: string, course_id: string): Promise<void>;
  markAsFinished(user_id: string, course_id: string): Promise<void>;
  getUserEnrollments(user_id: string): Promise<Matriculate[]>;
}

File: android/app/src/main/java/com/paralearn/android/domain/repository/MatriculateRepository.kt
package com.paralearn.android.domain.repository

import com.paralearn.android.domain.model.Enrollment

interface MatriculateRepository {
    suspend fun enroll(userId: String, courseId: String): Result<Unit>
    suspend fun getEnrollments(userId: String): Result<List<Enrollment>>
    suspend fun unenroll(userId: String, courseId: String): Result<Unit>
    suspend fun markAsFinished(userId: String, courseId: String): Result<Unit>
}
